SandWorm/SandWormComponent.cs

```diff
@@ -138,209 +137,186 @@ protected override void SolveInstance(IGH_DataAccess DA)
+            // Initialize all arrays
+            int trimmedWidth = KinectController.depthWidth - leftColumns - rightColumns;
+            int trimmedHeight = KinectController.depthHeight - topRows - bottomRows;
+            pointCloud = new Point3d[trimmedWidth * trimmedHeight];
+            int[] depthFrameDataInt = new int[trimmedWidth * trimmedHeight];
+            double[] averagedDepthFrameData = new double[trimmedWidth * trimmedHeight];
 
-                    Point3d tempPoint = new Point3d();
-                    Core.PixelSize depthPixelSize = Core.GetDepthPixelSpacing(sensorElevation);
+            // Initialize outputs
+            outputMesh = new List<Mesh>();
 
-                    // trim the depth array and cast ushort values to int
-                    Core.CopyAsIntArray(KinectController.depthFrameData, depthFrameDataInt, leftColumns, rightColumns, topRows, bottomRows, KinectController.depthHeight, KinectController.depthWidth);
+            Point3d tempPoint = new Point3d();
+            Core.PixelSize depthPixelSize = Core.GetDepthPixelSpacing(sensorElevation);
 
-                    averageFrames = averageFrames < 1 ? 1 : averageFrames; //make sure there is at least one frame in the render buffer
+            // Trim the depth array and cast ushort values to int
+            Core.CopyAsIntArray(KinectController.depthFrameData, depthFrameDataInt, leftColumns, rightColumns, topRows, bottomRows, KinectController.depthHeight, KinectController.depthWidth);
 
-                    // reset everything when resizing Kinect's field of view or changing the amounts of frame to average across
-                    if (renderBuffer.Count > averageFrames || quadMesh.Faces.Count != (trimmedWidth - 2) * (trimmedHeight - 2))
-                    {
-                        renderBuffer.Clear();
-                        Array.Clear(runningSum, 0, runningSum.Length);
-                        renderBuffer.AddLast(depthFrameDataInt);
-                    }
-                    else
-                        renderBuffer.AddLast(depthFrameDataInt);
+            averageFrames = averageFrames < 1 ? 1 : averageFrames; // Make sure there is at least one frame in the render buffer
 
-                    output.Add("Render buffer length:".PadRight(30, ' ') + renderBuffer.Count + " frames"); //debugging
+            // Reset everything when resizing Kinect's field of view or changing the amounts of frame to average across
+            if (renderBuffer.Count > averageFrames || quadMesh.Faces.Count != (trimmedWidth - 2) * (trimmedHeight - 2))
+            {
+                renderBuffer.Clear();
+                Array.Clear(runningSum, 0, runningSum.Length);
+                renderBuffer.AddLast(depthFrameDataInt);
+            }
+            else
+            { 
+                renderBuffer.AddLast(depthFrameDataInt);
+            }
+            Core.LogTiming(ref output, timer, "Render buffer length"); // Debug Info
```

Comment:
This one doesn't make sense anymore. The initial idea was to report on the amount of frames stored in the render buffer. The time spent on averaging is calculated and displayed in line 214